### DISCUSSION TITLE
Purpose of autoflush in example

### DISCUSSION THREAD

**Discussion by badarsebard** [fastapi/fastapi#5482] [Questions] [answered]
In the [SQL (Relational) Databases](https://fastapi.tiangolo.com/tutorial/sql-databases/) tutorial, the session factory is created using the SQLAlchemy `sessionmaker` function and called with the following arguments:
```python
SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
```
The default for the `autocommit` argument is already `False` so there's nothing there except being explicit to the user. However, the default for `autoflush` is `True`. Is there a specific reason for this? Is it best practice within FastAPI to configure it this way? What behavior might I expect if I use the default?

**Comment by tiangolo** [comment, marked as answer]
I have to update all that tutorial to use SQLModel, and a lot of that won't apply... For now, you can check the SQLModel docs.
